rfps/rfp-consensus-visualization-tool.md

```diff
@@ -0,0 +1,123 @@
+
+Regarding forks in EC, from the Filecoin Specification:
+
+> **It is possible for forks to emerge naturally in Expected Consensus.** Just as there can be zero miners winning a round, multiple miners can be elected as leaders in a given round. **This in turn means multiple blocks can be created per round.** In order to avoid wasting valid work done by miners, EC makes use of all valid blocks generated.
+>
+> Therefore the Filecoin chain can contain multiple valid blocks at each new chain height. All valid blocks in a round form what is called a **Tipset**. This greatly increases chain throughput by allowing blocks to propagate through the network of nodes more efficiently.
+>
+> Blocks from a given round are assembled into Tipsets according to certain rules (they must share the same parents and have been mined at the same height). Different miners may mine on different Tipsets because of network propagation delay.
+>
+> [More on Tipsets in the Spec...](https://filecoin-project.github.io/specs/#tipsets)
+
+Expected Consensus based on Storage Power Mining is a very exciting consensus algorithm in Filecoin. We would like to educate those new to Filecoin on how it works. We are also interested in observing when bad forks happen and probe when and why they may be occuring.
+
+#### Related Tools
+
+- A great example of an interactive tool that shows what is happening on the Ethereum chain is [https//ethviewer.live](http://ethviewer.live/).
+- A cool live blockchain sonification and visualization tool from Aleth.io can be seen at [https://audiokitpro.com/the-sound-of-blockchain](https://audiokitpro.com/the-sound-of-blockchain/).
+- There are also explorers that show information about uncles and chain reorgs such as [https://etherscan.io/blocks_forked](https://etherscan.io/blocks_forked).
+
+> (If you have other ideas for visualizations or tools that you think would be helpful for observing Filecoin consensus let us know!)
+
+## Deliverables
+
+An open-sourced, dual-licensed (under MIT and APACHE2) implementation that:
+
+- provides an interactive visualization of Filecoin consensus viewable in web browsers
+- or alternatively, a proposal to extend and enhance the existing explorer at [http://user.kittyhawk.wtf:8000](http://user.kittyhawk.wtf:8000/) ([Github repo](https://github.com/filecoin-project/filecoin-explorer))
+- displays new messages, blocks, tipsets, ancestors, forks, miners and related information via live updates
+- visualizes what is happening as messages are pooled into blocks forming Tipsets that extend Parent and GrandParent Tipsets as consensus is formed or forks occur
+- shows when natural and bad forks occur
+- provides explainers for what is happening and a glossary of terms
+
+
+And also:
+
+- has a well-documented, human-readable codebase
+- works with a Filecoin node API (local or remote) and database or cache if needed to perform the intended workflow
```

Comment:
Let's specify that this should work with lotus and/or go-filecoin (with a pref for lotus)

Reply:
is there a reason for a pref for lotus?